src/canvas-player-controls.js

```diff
@@ -0,0 +1,115 @@
+import videojs from 'video.js';
+/**
+ * This class reacts to interactions with the canvas and
+ * triggers appropriate functionality on the player. Right now
+ * it does two things:
+ *
+ * 1. A `mousedown`/`touchstart` followed by `touchend`/`mouseup` without any
+ *    `touchmove` or `mousemove` toggles play/pause on the player
+ * 2. Only moving on/clicking the control bar or toggling play/pause should
+ *    show the control bar. Moving around the scene in the canvas should not.
+ */
+class CanvasPlayerControls extends videojs.EventTarget {
+  constructor(player, canvas) {
+    super();
+
+    this.player = player;
+    this.canvas = canvas;
+
+    this.onMoveEnd = videojs.bind(this, this.onMoveEnd);
+    this.onMoveStart = videojs.bind(this, this.onMoveStart);
+    this.onMove = videojs.bind(this, this.onMove);
+    this.onControlBarMove = videojs.bind(this, this.onControlBarMove);
+
+    this.player.controlBar.on([
+      'mousedown',
+      'mousemove',
+      'mouseup',
+      'touchstart',
+      'touchmove',
+      'touchend'
+    ], this.onControlBarMove);
+
+    // we have to override these here because
+    // video.js listens for user activity on the video element
+    // and makes the user active when the mouse moves.
+    // We don't want that for 3d videos
+    this.oldReportUserActivity = this.player.reportUserActivity;
+    this.player.reportUserActivity = () => {};
+
+    // canvas movements
+    this.canvas.addEventListener('mousedown', this.onMoveStart);
+    this.canvas.addEventListener('touchstart', this.onMoveStart);
+    this.canvas.addEventListener('mousemove', this.onMove);
+    this.canvas.addEventListener('touchmove', this.onMove);
+    this.canvas.addEventListener('mouseup', this.onMoveEnd);
+    this.canvas.addEventListener('touchend', this.onMoveEnd);
+
+    this.shouldTogglePlay = false;
+  }
+
+  togglePlay() {
+    if (this.player.paused()) {
+      this.player.play();
+    } else {
+      this.player.pause();
+    }
+  }
+
+  onMoveStart(e) {
+
+    // if the player does not have a controlbar or
```

Comment:
Can this use Video.js's `tap` event?

Reply:
let me see if that works, looks like its probably a better solution.

Reply:
OK so I think that it would be too large of a refactor to convert this to a component and use the tap event. I think this will have to happen in another pr.

Reply:
sounds good.